[DEVX-161] Improve finance api docs (#118)
iZettle/api-documentation: finance-api/user-guides/fetch-account-transactions.md

<!-- Is there any limit for how old transactions can be fetched? -->  

* [Prerequisites](#prerequisites)
* [Fetch transactions during a specific period](#fetch-transactions-during-a-specific-period)
* [Fetch transactions of certain types during a specific period](#fetch-transactions-of-certain-types-during-a-specific-period)
* [Related task](#related-task)
* [Related API reference](#related-api-reference)

## Prerequisites
* Make sure that authorization is set up with the required OAuth scope using [Authorization OAuth2 API](../../authorization.adoc). 

## Fetch transactions during a specific period
To fetch transactions from a merchant's Zettle account during a specific period, you need to fetch them from their Zettle liquid account.
<!-- Check with team Ledger: Is it correct that a PAYMENT transaction can have a PAYOUT state? When I fetched all transactions, I got transactions that include PAYOUT. How can the integrator tell which PAYOUT is from which transaction type? --> 

1. Fetch transactions from the merchant's Zettle liquid account with pagination.
     
   ```

>    GET /organizations/self/accounts/{accountTypeGroup}/transactions?start={start_time}&end={end_time}&{limit}&{offset}

Reviewer: `limit` and `offset` parameters have names too

```suggestion
   GET /organizations/self/accounts/{accountTypeGroup}/transactions?start={start_time}&end={end_time}&limit={limit_value}&offset={offset_value}
```

   ```

   Example:
   

>    The following example fetches all transactions from the merchant's Zettle liquid account from 1 January, 2020 to 31 December, 2020. The requests fetch three transactions at a time, with pagination set by `limit` and `offset`. 

Reviewer: ```suggestion
   The following example fetches all transactions from the merchant's Zettle liquid account from 1 January, 2020 to 31 December, 2020. Each request fetches three transactions at a time, with pagination set by `limit` and `offset`. 
```

   
   Request
   
   ```
   GET /organizations/self/accounts/liquid/transactions?start=2020-01-01&end=2020-12-31&limit=3&offset=0
   ```
   ```
   GET /organizations/self/accounts/liquid/transactions?start=2020-01-01&end=2020-12-31&limit=3&offset=3
   ```

>    Response

Reviewer: How come one response for two requests?
And if `limit` is 3  why there are 6+ transactions in the response?

   
    ```json
    {
        "data": [
            {

>                 "timestamp": "2020-12-04T04:00:07.285+0000",

Reviewer: To be honest i miss the examples of `CARD_REFUND` and `CARD_PAYMENT_FEE_REFUND`, they are a bit special.
If limited info is present in the table of types then a solid example is needed.

            },
            {
                "timestamp": "2020-11-21T04:00:15.704+0000",
                "amount": -621,
                "originatorTransactionType": "CARD_PAYMENT_FEE",
                "originatingTransactionUuid": "6820265b-953e-43a7-bb65-abac1ef104bf"
            },
            {
                "timestamp": "2020-11-21T04:00:15.697+0000",
                "amount": 1100,
                "originatorTransactionType": "CARD_PAYMENT",
                "originatingTransactionUuid": "6820265b-953e-43a7-bb65-abac1ef104bf"
            },
            {
                "timestamp": "2020-09-10T09:51:35.162+0000",
                "amount": 5925,
                "originatorTransactionType": "FAILED_PAYOUT",
                "originatingTransactionUuid": "d8550d7a-f347-11ea-9612-3bce5300b9a9"
            },
            {
                "timestamp": "2020-09-10T09:27:28.590+0000",
                "amount": -5925,
                "originatorTransactionType": "PAYOUT",
                "originatingTransactionUuid": "d8550d7a-f347-11ea-9612-3bce5300b9a9"
            },
            ...
        ]
    }
    ```

> ## Step 2: Fetch transactions of certain types during a specific period

Reviewer 2: Having this as "step 2" seems that the user _must_ do the previous step in order to make this step works ... I am not sure if they are correlated to be called "steps"... maybe use the term "scenario", "example" or something like this?

Reviewer: Totally agree, those shouldn't be steps. They are independent from each other.

To fetch transactions of certain types from a merchant's Zettle account during a specific period, you need to fetch them from their Zettle liquid account.

For example, you can fetch all card transactions.

> 1. Fetch transactions of certain types from the merchant's Zettle liquid account. In the request path, specify transaction types as you need. See [supported transction types](../api-reference.md#supported-transaction-types).

Reviewer 2: typo in `transction `

Author: Thanks! Fixed now.

   > **Tip:** You specify the organisation UUID as `self`.
        
   ```
   GET /organizations/{organizationUuid}/accounts/liquid/transactions?{start}&{end}&{includeTransactionType}
   ```

   Example:
   

>    The following example request fetches all card payments and the associated card payment fees from the merchant's Zettle liquid account from 1 January, 2020 to 7 June, 2021.

Reviewer 2: you say jan `2020`, but the request has `2021`

Author: Changed it to 2020 now.

   
   ```

>    GET /organizations/self/accounts/liquid/transactions?start=2021-01-01&end=2021-06-07includeTransactionType=CARD_PAYMENT&includeTransactionType=CARD_PAYMENT_FEE

Reviewer 2: I think is missing a `&` after the end date?

Fetch account transactions
===
Using the Finance API, you can fetch all transactions or transactions of certain types from a merchant's Zettle liquid account during a specific period.

> **Note**: A merchant's Zettle liquid account contains all confirmed transactions that are already paid out or to be paid out to the merchant's bank account. If you want to check preliminary transactions that are being confirmed with buyers' banks, you can fetch them from a merchant's Zettle preliminary account.

<!-- Is there any limit for how old transactions can be fetched? -->  

* [Prerequisites](#prerequisites)
* [Step 1: Fetch all transactions during a specific period](#step-1-fetch-all-transactions-during-a-specific-period)
* [Step 2: Fetch transactions of certain types during a specific period](#step-2-fetch-transactions-of-certain-types-during-a-specific-period)
* [Related task](#related-task)
* [Related API reference](#related-api-reference)

## Prerequisites
* Make sure that authorization is set up using [Authorization OAuth2 API](../../authorization.adoc). 
<!-- to be continued if any -->

> ## Step 1: Fetch all transactions during a specific period

Reviewer: To me it's contradicting. When i fetch transactions for a specific period means that i don't fetch all of them only a selection. 
Suggestion:
 _Fetch all types of transactions during a specific period_
 
 Or 
 
 _Fetch transactions during a specific period_

To fetch all transactions from a merchant's Zettle account during a specific period, you need to fetch them from their Zettle liquid account.
<!-- Check with team Ledger: Is it correct that a PAYMENT transaction can have a PAYOUT state? When I fetched all transactions, I got transactions that include PAYOUT. How can the integrator tell which PAYOUT is from which transaction type? --> 

1. Optional: Fetch your organisation UUID. 
   > **Tip:** You don't need to fetch the organisation UUID, as you can specify the organisation UUID as `self` in requests that require it.

    ```
    GET users/me
    ```

>    Example:
>        
>    The following example response returns organisation UUID `ab305d54-75b4-431b-adb2-eb6b9e546013`.

Reviewer: ```suggestion
   Response example:
```

    ```
    {
        "uuid": "de305d54-75b4-431b-adb2-eb6b9e546014",
        "organizationUuid": "ab305d54-75b4-431b-adb2-eb6b9e546013"
    }
    ```

2. Fetch all transactions from the merchant's Zettle liquid account.
      > **Tip:** You specify the organisation UUID as `self`.
     
   ```
   GET /organizations/{organizationUuid}/accounts/liquid/transactions?{start}&{end}
   ```

   Example:
   

>    The following example request fetches all transactions from the merchant's Zettle liquid account from 1 January, 2020 to 7 June, 2021.

Reviewer: As mentioned in previous comments  word "all" should be used carefully. In this case we are not fetching all transactions just a selection "from 1 January, 2020 to 7 June, 2021"

   
   ```
   GET /organizations/self/accounts/liquid/transactions?start=2021-01-01&end=2021-06-07
   ```
       

>    The following example response returns all the transactions from 1 January, 2020 to 7 June, 2021.

Reviewer: Responses do not return anything, they are returned by REST APIs.
Can  you just write  a line "Example response" or  "Example of successful response".
I feel it's over explained in many places, as a developer i would get overwhelmed with so much text.

            },
            {
                "timestamp": "2020-11-21T04:00:15.697+0000",
                "amount": 1100,
                "originatorTransactionType": "CARD_PAYMENT",
                "originatingTransactionUuid": "6820265b-953e-43a7-bb65-abac1ef104bf"
            },
            {
                "timestamp": "2020-09-10T09:51:35.162+0000",
                "amount": 5925,
                "originatorTransactionType": "FAILED_PAYOUT",
                "originatingTransactionUuid": "d8550d7a-f347-11ea-9612-3bce5300b9a9"
            },
            {
                "timestamp": "2020-09-10T09:27:28.590+0000",
                "amount": -5925,
                "originatorTransactionType": "PAYOUT",
                "originatingTransactionUuid": "d8550d7a-f347-11ea-9612-3bce5300b9a9"
            },
            ...
        ]
    }
    ```

## Step 2: Fetch transactions of certain types during a specific period
To fetch transactions of certain types from a merchant's Zettle account during a specific period, you need to fetch them from their Zettle liquid account.

For example, you can fetch all card transactions.

> 1. Fetch transactions of certain types from the merchant's Zettle liquid account. In the request path, specify transaction types as you need. See [supported transaction types](../api-reference.md#supported-transaction-types).

Reviewer: > In the request path, specify transaction types as you need.

Transaction types can only be specified as query parameters not in the path.

   > **Tip:** You specify the organisation UUID as `self`.
        
   ```
   GET /organizations/{organizationUuid}/accounts/liquid/transactions?{start}&{end}&{includeTransactionType}
   ```

   Example:
   
   The following example request fetches all card payments and the associated card payment fees from the merchant's Zettle liquid account from 1 January, 2020 to 7 June, 2021.
   
   ```

>    GET /organizations/self/accounts/liquid/transactions?start=2020-01-01&end=2021-06-07&includeTransactionType=CARD_PAYMENT&includeTransactionType=CARD_PAYMENT_FEE

Reviewer: I think this example is bad, we don't want partners to fetch transactions for 1,5 year since this kind of requests can time out. 
Of course they can try if they need to. But usually an accounting period is 1 year.

   ```
       

>    The following example response returns a card payment and the associated card payment fee that happened during the period from 1 January, 2020 to 7 June, 2021.

Reviewer: ```suggestion
   Example response:
```

    ```json
    {
        "data": [
            {
                "timestamp": "2020-11-21T04:00:15.704+0000",
                "amount": -621,
                "originatorTransactionType": "CARD_PAYMENT_FEE",
                "originatingTransactionUuid": "6820265b-953e-43a7-bb65-abac1ef104bf"
            },
            {
                "timestamp": "2020-11-21T04:00:15.697+0000",
                "amount": 1100,
                "originatorTransactionType": "CARD_PAYMENT",
                "originatingTransactionUuid": "6820265b-953e-43a7-bb65-abac1ef104bf"

>             },

Reviewer: If the intention here is to show shortened example response we need to add 3 dots `...` after the last comma in the array, otherwise it looks like a broken json.

Fetch account transactions
===
Use the Finance API to fetch transactions or transactions of certain types from a merchant's liquid account during a specific period.

> > **Note**: A merchant's liquid account contains all confirmed transactions that are already paid out or to be paid out to the merchant. If you want to check preliminary transactions that are being confirmed with the issuing banks (buyers' banks), you can fetch them from a merchant's preliminary account.

Reviewer 3: Same thing here.
```
If you want to check preliminary transactions that are being confirmed with the issuing banks (buyers' banks), you can fetch them from a merchant's preliminary account.
```
this is not correct.

Reviewer 3: ```
A merchant's liquid account contains all confirmed transactions that are already paid out or to be paid out to the merchant. 
```

It shouldn't contain transactions that are already paid out.

   ```

   Example:
   
   The following example fetches the latest three transactions from the merchant's liquid account from 1 January, 2020 to 5 July, 2020. 
   
   Request   
   ```
   GET /organizations/self/accounts/liquid/transactions?start=2020-01-01&end=2020-07-05&limit=3&offset=0
   ```
   Response         
   ```json
       {
           "data": [
               {
                   "timestamp": "2020-07-04T20:16:44.309+0000",
                   "amount": 381,
                   "originatorTransactionType": "CARD_PAYMENT_FEE_REFUND",
                   "originatingTransactionUuid": "30cef6e2-be09-11ea-a8e4-bce028663c34"
               },
               {
                   "timestamp": "2020-07-04T20:16:44.309+0000",
                   "amount": -20610,
                   "originatorTransactionType": "CARD_REFUND",
                   "originatingTransactionUuid": "30cef6e2-be09-11ea-a8e4-bce028663c34"
               },
               {
                   "timestamp": "2020-06-27T23:52:18.327+0000",
                   "amount": 649,

>                    "originatorTransactionType": "PAYMENT_FEE", // In case of refund

Reviewer 3: What does `PAYMENT_FEE` has to do with refund?

Reviewer: This would  probably be better like this to show that amount for fee is positive in case of refund
{
                   "timestamp": "2020-06-27T23:52:18.327+0000",
                   "amount": 649, // Positive in case of refund
                   "originatorTransactionType": "PAYMENT_FEE",

                   "originatingTransactionUuid": "690c99ea-b6ef-11ea-9730-7ef7aeff642d"
               },
           ]
       }
   ```
   
2. Send another request where you keep `limit` the same as in step 1 and increment `offset` with the value of `limit` . 
     
   ```
   GET /organizations/self/accounts/{accountTypeGroup}/transactions?start={start_time}&end={end_time}&{limit}={limit_value}&offset={offset_value}
   ```
   Example:
      
   The following example fetches the next three transactions from the merchant's liquid account from 1 January, 2020 to 5 July, 2020.
   
   Request
    
   ```
   GET /organizations/self/accounts/liquid/transactions?start=2020-01-01&end=2020-07-05&limit=3&offset=3
   ```
   Response
   
    ```json
    {
        "data": [
            {
                "timestamp": "2020-06-27T23:52:18.327+0000",
                "amount": -35100,

>                 "originatorTransactionType": "PAYMENT", // In case of refund

Reviewer 3: I didn't get this example. Are you trying to say that if the payment `45f51ed4-b7bf-11ea-90de-435a3f6e4738` was refunded we would show this?

Reviewer: This would probably be better like this to show that amount is negative in case of refund
 {
                "timestamp": "2020-06-27T23:52:18.327+0000",
                "amount": -35100, // Negative in case of refund
                "originatorTransactionType": "PAYMENT",
